primitives/api/src/lib.rs

```diff
@@ -397,7 +398,7 @@ pub trait ConstructRuntimeApi<Block: BlockT, C: CallApiAt<Block>> {
 #[cfg(feature = "std")]
 pub trait ApiErrorExt {
 	/// Error type used by the runtime apis.
-	type Error: std::fmt::Debug + From<String>;
+	type Error: std::fmt::Debug + From<(&'static str, codec::Error)>;
```

Comment:
Please introduce a custom error type here in the api crate.

Reply:
I was thinking about this, the more it spread. Will do